src/artemis/ispyb/store_in_ispyb.py

```diff
@@ -30,6 +31,38 @@ def __init__(self, ispyb_config):
         self.mx_acquisition = None
         self.core = None
 
+        self.datacollection_ids = None
+        self.datacollection_group_id = None
+        self.grid_ids = None
+
+    def __enter__(self):
+        (
+            self.datacollection_ids,
+            self.grid_ids,
+            self.datacollection_group_id,
+        ) = self.store_grid_scan(self.full_params)
+        for id in self.datacollection_ids:
+            run_start(id)
+
+    def __exit__(self, exception, exception_value, traceback):
+        if exception is not None:
+            run_status = "DataCollection Unsuccessful"
+        else:
+            run_status = "DataCollection Successful"
+        current_time = self.get_current_time_string()
+        for id in self.datacollection_ids:
+            self.update_grid_scan_with_end_time_and_status(
+                current_time,
+                run_status,
+                id,
+                self.datacollection_group_id,
+            )
+
+        for id in self.datacollection_ids:
+            run_end(id)
+
+        wait_for_result(self.datacollection_group_id)
```

Comment:
*Must* - We don't assign the value `wait_for_result` returns on the main branch at the moment, but we will need the value when we want to move to the position analysis gives us. I would suggest returning the ids from the __enter__ method and assigning them to variables in the plan. So it would look something like:

```
with ispyb as ids:
        dc_ids = ids[0]
        dcg_id = id[2]
        run_start(dc_ids)
        do_fgs()

run_end(dc_ids)
wait_for_results(dcg_id)
```